tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/DLQAfterExpiredMessageTest.java

```diff
@@ -0,0 +1,170 @@
+
+      // Default DLQ
+      server.addAddressInfo(new AddressInfo(SimpleString.toSimpleString(getDeadLetterAddress()), RoutingType.ANYCAST));
+      server.createQueue(new QueueConfiguration(getDeadLetterAddress()).setRoutingType(RoutingType.ANYCAST));
+
+      // Expiry
+      server.addAddressInfo(new AddressInfo(SimpleString.toSimpleString(getExpiryQueue()), RoutingType.ANYCAST));
+      server.createQueue(new QueueConfiguration(getExpiryQueue()).setRoutingType(RoutingType.ANYCAST));
+
+      // Default Topic
+      server.addAddressInfo(new AddressInfo(SimpleString.toSimpleString(getTopicName()), RoutingType.MULTICAST));
+      server.createQueue(new QueueConfiguration(getTopicName()));
+
+      // Additional Test Queues
+      for (int i = 0; i < getPrecreatedQueueSize(); ++i) {
+         server.addAddressInfo(new AddressInfo(SimpleString.toSimpleString(getQueueName(i)), RoutingType.ANYCAST));
+         server.createQueue(new QueueConfiguration(getQueueName(i)).setRoutingType(RoutingType.ANYCAST));
+      }
+   }
+
+   @Override
+   protected void configureAddressPolicy(ActiveMQServer server) {
+      // Address configuration
+      AddressSettings addressSettings = new AddressSettings();
+
+      addressSettings.setAddressFullMessagePolicy(AddressFullMessagePolicy.PAGE);
+      addressSettings.setAutoCreateQueues(isAutoCreateQueues());
+      addressSettings.setAutoCreateAddresses(isAutoCreateAddresses());
+      addressSettings.setDeadLetterAddress(SimpleString.toSimpleString(getDeadLetterAddress()));
+      addressSettings.setExpiryAddress(SimpleString.toSimpleString(getExpiryQueue()));
+      addressSettings.setMaxDeliveryAttempts(1);
+      server.getConfiguration().getAddressesSettings().put("#", addressSettings);
+      server.getConfiguration().getAddressesSettings().put(getExpiryQueue(), addressSettings);
+   }
+
+   @Test
+   public void testDoubleTransfer() throws Throwable {
+
+      AmqpClient client = createAmqpClient();
+      AmqpConnection connection = addConnection(client.connect());
+      try {
+         AmqpSession session = connection.createSession();
+
+         AmqpSender sender = session.createSender(getQueueName());
+
+         // Get the Queue View early to avoid racing the delivery.
+         final Queue queueView = getProxyToQueue(getQueueName());
+         assertNotNull(queueView);
+
+         AmqpMessage message = new AmqpMessage();
+         message.setTimeToLive(1);
+         message.setText("Test-Message");
+         message.setDurable(true);
+         message.setApplicationProperty("key1", "Value1");
+         sender.send(message);
+         sender.close();
+
+         Wait.assertEquals(1, queueView::getMessagesExpired);
+
+         // Now try and get the message
+         AmqpReceiver receiver = session.createReceiver(getQueueName());
+         receiver.flow(1);
+         AmqpMessage received = receiver.receiveNoWait();
```

Comment:
It would eventually fail if there's a message.

I have been using as a rule, if the expected is no delievery, it should asserNull(receiveNoWait());

There's already an assertion before on the number of expires.

Reply:
I don't need that here... I'm already asserting before.

I was just explaining why I prefer using receiveNoWait();

I had some major cleanup on the testsuite weeks ago where I fixed a lot of receive(1000); that was intended to noWait.